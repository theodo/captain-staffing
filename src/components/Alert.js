import React from 'react'
import PropTypes from 'prop-types'

export default class Alert extends React.Component {

  static propTypes = {
    error: PropTypes.object.isRequired,
  }

  render() {
    let message
    let icon

    switch (this.props.error.code) {
      case 403:
        icon = '⛔️'
        message = 'You don’t have permission to access this Spreadsheet.'
        break
      case 404:
        icon = '❓'
        message = 'Spreadsheet not found.'
        break
      case 503:
        icon = '⌚'
        message = 'Service unavailable.'
        break
      case 400:
        icon = '☹'
        message = this.props.error.message
        break
      default:
        icon = '💀'
        message = 'Doh, I couldn’t load the data.'
    }

    return (
      <p className="alert">
        <span className="alert__icon">{ icon }</span>
        <span className="alert__message">{ message }</span>
      </p>
    )
  }
}
