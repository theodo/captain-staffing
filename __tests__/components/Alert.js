import Alert from 'components/Alert'
import React from 'react'

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

describe('Alert', () => {
  it('should handle 403 error', () => {
    const error = { code: 403 }
    const alert = shallow(<Alert error={error} />)

    expect(toJSON(alert)).toMatchSnapshot()
  })

  it('should handle 404 error', () => {
    const error = { code: 404 }
    const alert = shallow(<Alert error={error} />)

    expect(toJSON(alert)).toMatchSnapshot()
  })

  it('should handle 400 error', () => {
    const error = { code: 400, message: 'error' }
    const alert = shallow(<Alert error={error} />)

    expect(toJSON(alert)).toMatchSnapshot()
  })

  it('should handle 503 error', () => {
    const error = { code: 503 }
    const alert = shallow(<Alert error={error} />)

    expect(toJSON(alert)).toMatchSnapshot()
  })

  it('should handle default error', () => {
    const error = { code: 0 }
    const alert = shallow(<Alert error={error} />)

    expect(toJSON(alert)).toMatchSnapshot()
  })
})
