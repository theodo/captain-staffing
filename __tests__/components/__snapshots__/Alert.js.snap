exports[`Alert should handle 400 error 1`] = `
<p
  className="alert">
  <span
    className="alert__icon">
    ☹
  </span>
  <span
    className="alert__message">
    error
  </span>
</p>
`;

exports[`Alert should handle 403 error 1`] = `
<p
  className="alert">
  <span
    className="alert__icon">
    ⛔️
  </span>
  <span
    className="alert__message">
    You don’t have permission to access this Spreadsheet.
  </span>
</p>
`;

exports[`Alert should handle 404 error 1`] = `
<p
  className="alert">
  <span
    className="alert__icon">
    ❓
  </span>
  <span
    className="alert__message">
    Spreadsheet not found.
  </span>
</p>
`;

exports[`Alert should handle 503 error 1`] = `
<p
  className="alert">
  <span
    className="alert__icon">
    ⌚
  </span>
  <span
    className="alert__message">
    Service unavailable.
  </span>
</p>
`;

exports[`Alert should handle default error 1`] = `
<p
  className="alert">
  <span
    className="alert__icon">
    💀
  </span>
  <span
    className="alert__message">
    Doh, I couldn’t load the data.
  </span>
</p>
`;
